portal/static/js/accountCreation.js

```diff
@@ -0,0 +1,572 @@
+
+            var orgIDs = $("#userOrgs input:checked").map(function(){
+                return { reference: "api/organization/"+$(this).val() };
+            }).get();
+
+            _demoArray["careProvider"] = orgIDs;
+
+            var arrCommunication = OT.getCommunicationArray();
+            if (arrCommunication.length > 0) {
+                _demoArray["communication"] = arrCommunication;
+            };
+
+            /*** SYSTEM uri is defined by SYSTEM_IDENTIFIER_ENUM, see main.js for details **/
+            var studyId = $("#profileStudyId").val();
+            if (hasValue(studyId)) {
+                var studyIdObj = {
+                    system: SYSTEM_IDENTIFIER_ENUM["external_study_id"],
+                    use: "secondary",
+                    value: studyId
+                };
+                if (!_demoArray["identifier"]) {
+                    _demoArray["identifier"] = [];
+                };
+                _demoArray["identifier"].push(studyIdObj);
+            };
+
+            var siteId = $("#profileSiteId").val();
+            if (hasValue(siteId)) {
+                var siteIdObj = {
+                    system: SYSTEM_IDENTIFIER_ENUM["external_site_id"],
+                    use: "secondary",
+                    value: siteId
+                };
+                if (!_demoArray["identifier"]) _demoArray["identifier"] = [];
+                _demoArray["identifier"].push(siteIdObj);
+            };
+
+
+            var states = [];
+            $("#userOrgs input[name='organization']").each(function() {
+                if ($(this).is(":checked")) {
+                    if (hasValue($(this).attr("state"))) states.push($(this).attr("state"));
+                };
+            });
+
+            if (states.length > 0) {
+                if (!_demoArray["identifier"]) _demoArray["identifier"] = [];
+                states.forEach(function(state) {
+                    _demoArray["identifier"].push({
+                        system: SYSTEM_IDENTIFIER_ENUM["practice_region"],
+                        use: "secondary",
+                        value: "state:" + state
+                    });
+                });
+            };
+            self.__request({"apiUrl":"/api/demographics/"+this.userId, "requestType": "PUT", "requestData": JSON.stringify(_demoArray), "sync": true, "callback":
+                function(data){
+                    if (data.error) {
+                        self.__handleError(data.error);
+                        self.__handleButton();
+                    } else {
+                        self.__setProcedures();
+                    };
+                }
+            });
+        } else {
+            //Display Error Here
+            if (returnedData.error) {
+               this.__handleError(returnedData.error);
+               this.__handleButton();
+            };
+        };
+    };
+    this.__setProcedures = function() {
```

Comment:
Should this be moved to `procedures.js`?

Reply:
@ivan, I didn't fix this here because there is code in this method that acts on the UI specific to account creation page - hard to make it work if the method is moved to procedures.js.  Otherwise, I addressed your comments.   I didn't address some of issues raised by codacy-bot - e.g. hasValue not defined, etc.  as the undefined functions were actually defined in main.js (I did have a comment in the js file that there is a dependency on main.js)

Reply:
Also, Ivan, let me know if you see any issue with my latest changes.

Reply:
Ah, thanks for the explanation :)

Reply:
There isn't a way to pass those variables between files, is there (maybe as values to functions)? Seems like there should be a more robust way to know you won't have undefined variables.

Reply:
@ivan-c ivan, yeah, was thinking over the dependency issue here.. thinking to treat each JS file as a module and load each when needed via modular script loader like RequireJS (or CommonJS).  I will do more research into this.